lib/kafka_producer.js

```diff
@@ -181,16 +181,31 @@ KafkaProducer.prototype.batch = function batch(topic, message, timeStamp, callba
     }
 };
 
+// Only run the callback when all the batches are flushed
 KafkaProducer.prototype.flushEntireCache = function flushEntireCache(callback) {
     var self = this;
 
-    var handleResponse = function handleResponse(err, res) {
-        if (callback) {
-            callback(err, res);
+    var numToBeFlushed = 0;
+    var numFlushed = 0;
+
+    var keys = Object.keys(self.topicToBatchQueue);
```

Comment:
why not count on all the keys, and flush even if the batch is 0. Logic here will be much simpler.